File: src/modules/organization/services/organization-update-status.service.ts
import { Injectable, NotFoundException } from '@nestjs/common';
import { PrismaService } from 'src/infra/database/prisma.service';
import { UserRequest } from 'src/shared/types/user-request';
import { userOrganizationValidator } from 'src/shared/validators/user-organization.validator';
import { OrganizationUpdateStatusDto } from '../domain/dto/organization-update-status.dto';
import { OrganizationEntity } from '../domain/entities/organization.entity';

@Injectable()
export class OrganizationUpdateStatusService {
  constructor(private prisma: PrismaService) {}
  async execute(
    id: number,
    dto: OrganizationUpdateStatusDto,
    userRequest: UserRequest,
  ): Promise<OrganizationEntity> {
    const organization = await this.prisma.organization.findFirst({
      where: {
        id: id,
        deletedAt: null,
      },
    });

    if (!organization)
      throw new NotFoundException('Organização não encontrada!');

    await userOrganizationValidator(id, userRequest);

    const updatedOrganization = await this.prisma.organization.update({
      data: {
        status: dto.status,
      },
      where: {
        id: id,
      },
    });

    return new OrganizationEntity(updatedOrganization);
  }
}

File: src/modules/organization/services/organization_update_status_service.py
from src.shared.types.user_request import UserRequest
from src.modules.organization.models import Organization
from src.modules.organization.domain.dto.organization_update_status_dto import (
    OrganizationUpdateStatusDTO,
)
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.validators.user_organization_validator import (
    user_organization_validator,
)


class OrganizationUpdateStatusService:
    @classmethod
    def execute(
        self, id: int, dto: OrganizationUpdateStatusDTO, user_request: UserRequest
    ):

        organization = Organization.objects.filter(id=id, deleted_at=None).first()

        if organization is None:
            raise NotFoundException("Organização não encontrada!")

        user_organization_validator(id, user_request)

        organization.status = dto.status

        organization.save()

        return organization
